test/cpp/qps/server_async.cc

```diff
@@ -162,7 +162,10 @@ class AsyncQpsServerTest final : public grpc::testing::Server {
       std::lock_guard<std::mutex> lock((*ss)->mutex);
       (*ss)->shutdown = true;
     }
-    std::thread shutdown_thread(&AsyncQpsServerTest::ShutdownThreadFunc, this);
+    // TODO(vjpai): Remove the following deadline and allow full proper
+    // shutdown.
```

Comment:
What is the full proper shutdown you want to do later?

Reply:
Ideally we should be able to remove the deadline on it. The deadline prevents it from taking too long but, on the other hand, taking too long might be a signal that something bad is happening in the stack. Note that this is a pre-existing TODO; it just moved from the ShutdownThreadFunc where it was an async TODO to the destructor where it is a synchronously-executed TODO.

Reply:
Do you mind if we hold off on removing the TODO until a later PR, since it's orthogonal to the API misuse being fixed in this PR?

Reply:
Sure, I am fine with removing it later. Just wondering what is the next step.